### PR TITLE
Add coordinates display to the debug panel

### LINKED ISSUE #4151: Add an option to enable showMouseCoordinates for the hover panel
That variable was left to eventually allow re-enabling the showing on world coordinates in the mouse hover panel. It's not been done yet, so I'm opening this issue. The setting might be nice to have in the options menu for easy accessibility as it can be helpful in general debugging, not just when writing new code.

### PR DESCRIPTION
**Pull request by Kasterisk**
**Brief Description of What This PR Does**

I end up removing showMouseCoordinates completely in favor of displaying debug coordinates in the debug panel which should be more stage-neutral. There are now 2 coordinates indicator: player position and where the player is looking at (this should be relevant in most/all of stages).

**Related Issues**

Closes #4151.

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by 84634E1A607A** [comment]
This panel now looks a little bit strange... (Judging from the screenshot)

**Comment by Kasterisk** [comment]
That's to accommodate the Vector3 strings at full precision, I don't think we ought to make debugs pretty either way.

**Comment by hhyyrylainen** [comment]
I can think many more such numbers like view direction we'll need for the later game that maybe it'd make sense to dedicate an entire sub panel for numbers like this instead of putting in the main panel, which I think we'll have plenty filled with checkboxes for enabling various debug things.

File: Thrive.csproj [hunk 1]
```diff
@@ -232,7 +233,7 @@
     <Compile Include="src\gui_common\CompoundListBox.cs" />
     <Compile Include="src\gui_common\CustomCheckBox.cs" />
     <Compile Include="src\gui_common\CreditsScroll.cs" />
-    <Compile Include="src\gui_common\CustomDialog.cs" />
+    <Compile Include="src\gui_common\dialogs\CustomDialog.cs" />
```
Comment (Kasterisk) [inline Thrive.csproj:236]: I keep losing track of where this class is, so this should make it easier to find.

**Comment by hhyyrylainen** [comment]
Yeah, I think that's pretty good, we can put like extra info for the hovered object etc. in there in the future.

File: src/modding/NewModGUI.tscn [hunk 2]
```diff
@@ -1,9 +1,10 @@
 [gd_scene load_steps=4 format=2]
 
-[ext_resource path="res://src/gui_common/CustomDialog.tscn" type="PackedScene" id=1]
+[ext_resource path="res://src/gui_common/dialogs/CustomDialog.tscn" type="PackedScene" id=1]
 [ext_resource path="res://src/modding/NewModGUI.cs" type="Script" id=2]
 [ext_resource path="res://src/gui_common/fonts/Lato-Bold-AlmostSmaller.tres" type="DynamicFont" id=3]
 
+
```
Comment (hhyyrylainen) [inline src/modding/NewModGUI.tscn:7]: A bunch of extra newlines seem to get inserted. For me Godot doesn't add these in the scene files...

Comment (Kasterisk) [reply src/modding/NewModGUI.tscn:7]: I don't know, these were automatic. I noticed I'm running 3.5, latest is 3.5.1 right? miiight the cause. Should these be removed now or let them be?

Comment (hhyyrylainen) [reply src/modding/NewModGUI.tscn:7]: We are still using 3.5. Removing these would reduce the number of translation file changes *a lot*.

File: src/microbe_stage/MicrobeStage.cs [hunk 3]
```diff
@@ -192,9 +192,11 @@ public override void _Process(float delta)
 
         if (Player != null)
         {
-            var playerTransform = Player.GlobalTransform;
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeStage.cs:195]: I think the player transform variable should be kept as it is used in multiple places in a few lines.

File: src/microbe_stage/MicrobeStage.cs [hunk 4]
```diff
@@ -192,9 +192,11 @@ public override void _Process(float delta)
...
-            spawner.Process(delta, playerTransform.origin);
-            Clouds.ReportPlayerPosition(playerTransform.origin);
+            DebugOverlays.Instance.ReportPositionCoords(Player.GlobalTranslation);
+            DebugOverlays.Instance.ReportLookingAtCoords(Camera.CursorWorldPos);
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeStage.cs:196]: I'd prefer these to have the full "Coordinates" name in it, function names are not space constrained like the GUI where I think having "coords" in the debug view is fine.

File: src/engine/DebugOverlays.Inspector.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
+﻿using Godot;
+
+public partial class DebugOverlays
```
Comment (hhyyrylainen) [inline src/engine/DebugOverlays.Inspector.cs:3]: This partial class needs a summary comment here. The StyleCop warning should be on but I don't know why it doesn't trigger here.

**Comment by hhyyrylainen** [comment]
I now playtested this and this seems fine, except the label should have cut (or word wrap) on, and the inspector panel should be wider by default:

**Comment by hhyyrylainen** [comment]
I'll fix this myself so that I can get this merged now to make my own PR's translation changes on top of the ones in this.

**Comment by Kasterisk** [comment]
Ah, I just pushed my changes now.

**Comment by hhyyrylainen** [comment]
I added a scroll container in preparation for having other stuff there, I'll combine my changes with yours.

**Comment by hhyyrylainen** [comment]
I'll merge this if you are fine with my new changes.

**Comment by hhyyrylainen** [comment]
I'll merge this now to not block my work...

**Pull request merged**